server/polar/integrations/github/service/issue.py

```diff
@@ -550,154 +545,70 @@ async def sync_external_org_with_repo_and_issue(
         issue_number: int,
     ) -> Tuple[Organization, Repository, Issue]:
         log.info(
-            "syncing external",
+            "syncing external issue",
             org_name=org_name,
             repo_name=repo_name,
             issue_number=issue_number,
         )
 
-        organization = await github_organization.get_by_name(
-            session, Platforms.github, org_name
+        issue = await self.get_by_external_lookup_key(
+            session, Platforms.github, f"{org_name}/{repo_name}/{issue_number}"
         )
-
-        if not organization:
-            log.info("organization not found by name", org_name=org_name)
-
-            try:
-                repo_response = await client.rest.repos.async_get(org_name, repo_name)
-                github_repo = repo_response.parsed_data
-                owner = github_repo.owner
-                is_personal = owner.type.lower() == "user"
-            except RequestFailed as e:
-                if e.response.status_code == 404:
-                    raise ResourceNotFound()
-                if e.response.status_code == 401:
-                    raise ResourceNotFound()
-                # re-raise other status codes
-                raise e
-
-            # check if we have org with same external_id
-            organization = await github_organization.get_by_external_id(
-                session, owner.id
-            )
-
-        # still no organization, create it
-        if not organization:
-            log.info(
-                "organization not found by external_id, creating it",
+        if issue is not None:
+            log.debug(
+                "external issue found by lookup key",
                 org_name=org_name,
-                external_id=owner.id,
-            )
-
-            organization = await github_organization.create(
-                session,
-                OrganizationCreate(
-                    platform=Platforms.github,
-                    name=owner.login,
-                    external_id=owner.id,
-                    avatar_url=owner.avatar_url,
-                    is_personal=is_personal,
-                ),
-            )
-
-        repository = await github_repository.get_by_org_and_name(
-            session,
-            organization.id,
-            repo_name,
-        )
-
-        if not repository:
-            log.info(
-                "repository not found by name",
-                organization_id=organization.id,
                 repo_name=repo_name,
+                issue_number=issue_number,
+                external_id=issue.external_id,
             )
+            issue = await self.get_loaded(session, issue.id)
+            assert issue is not None
+            return (issue.repository.organization, issue.repository, issue)
```

Comment:
I just noticed that while this function returns `Tuple[Organization, Repository, Issue]`, the `Issue` is the only field that ever is used. A good idea for some cleanup  😄  (no need to do it in this PR)

Reply:
Yes, I saw they were not used, but unsure if it was on purpose or not. Will do a follow-up PR to remove this.